[new release] current (10 packages) (0.6.5)

benmandrew opened the pull request:
Pipeline language for keeping things up-to-date

- Project page: <a href="https://github.com/ocurrent/ocurrent">https://github.com/ocurrent/ocurrent</a>
- Documentation: <a href="https://ocurrent.github.io/ocurrent/">https://ocurrent.github.io/ocurrent/</a>

##### CHANGES:

API:

- Add `Current_term.observe` to evaluate the current state of a term. (@TheLortex, ocurrent/ocurrent#374)
- Add `up_args` to `Current_docker.compose_cli` (@maiste, ocurrent/ocurrent#418)
- Add support for `buildx` in `Current_docker.build` (@maiste, ocurrent/ocurrent#418)
- Add `auth` to `Current_docker.pull` (@maiste, ocurrent/ocurrent#423)
- Add ability to directly construct `Current_git.Commit.t` values (@ElectreAAS, ocurrent/ocurrent#442)

Plugins:

- GitHub: Fetch most relevant PRs and branches by update dates (@punchagan, ocurrent/ocurrent#421)
- Docker: Add Prometheus metrics for Docker operations (@benmandrew, ocurrent/ocurrent#430 ocurrent/ocurrent#437 ocurrent/ocurrent#440)
- Git: Add Prometheus metrics for `git clone`, `git fetch`, and `git submodule update` (@benmandrew, ocurrent/ocurrent#430)

Other:

- Bump constraint to OCaml 4.12. (@MisterDA, ocurrent/ocurrent#415)

raphael-proust commented:
@benmandrew  There seem to be some issues with the code that make CI hard fail (and the packages likely uninstallable):

```
[ERROR] The compilation of current_gitlab.0.6.5 failed at "dune build -p current_gitlab -j 255 @install".

#=== ERROR while compiling current_gitlab.0.6.5 ===============================#
# context              2.2.0~beta1 | linux/x86_64 | ocaml-base-compiler.4.13.1 | pinned(https://github.com/ocurrent/ocurrent/releases/download/v0.6.5/current-0.6.5.tbz)
# path                 ~/.opam/4.13/.opam-switch/build/current_gitlab.0.6.5
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p current_gitlab -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/current_gitlab-7-ca6c5f.env
# output-file          ~/.opam/log/current_gitlab-7-ca6c5f.out
### output ###
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlc.opt -w -40 -g -bin-annot -I plugins/gitlab/.current_gitlab.objs/byte -I /home/opam/.opam/4.13/lib/ISO8601 -I /home/opam/.opam/4.13/lib/angstrom -I /home/opam/.opam/4.13/lib/ansi -I /home/opam/.opam/4.13/lib/asetmap -I /home/opam/.opam/4.13/lib/astring -I /home/opam/.opam/4.13/lib/atdgen -I /home/opam/.opam/4.13/lib/atdgen-runtime -I /home/opam/.opam/4.13/lib/base/caml -I /home/opam/.opam/4.13/lib/base64 -I /home/opam/.opam/4.13/lib/base64/rfc2045 -I /home/opam/.opam/4.13/lib/bigarray-overlap -I /home/opam/.opam/4.13/lib/bigstringaf -I /home/opam/.opam/4.13/lib/biniou -I /home/opam/.opam/4.13/lib/bos -I /home/opam/.opam/4.13/lib/bytes -I /home/opam/.opam/4.13/lib/camlp-streams -I /home/opam/.opam/4.13/lib/cmdliner -I /home/opam/.opam/4.13/lib/cohttp -I /home/opam/.opam/4.13/lib/cohttp-lwt -I /home/opam/.opam/4.13/lib/cohttp-lwt-unix -I /home/opam/.opam/4.13/lib/conduit -I /home/opam/.opam/4.13/lib/conduit-lwt -I /home/opam/.opam/4.13/lib/conduit-lwt-unix -I /home/opam/.opam/4.13/lib/cstruct -I /home/opam/.opam/4.13/lib/csv -I /home/opam/.opam/4.13/lib/current -I /home/opam/.opam/4.13/lib/current/cache -I /home/opam/.opam/4.13/lib/current/term -I /home/opam/.opam/4.13/lib/current_git -I /home/opam/.opam/4.13/lib/current_incr -I /home/opam/.opam/4.13/lib/current_web -I /home/opam/.opam/4.13/lib/domain-name -I /home/opam/.opam/4.13/lib/duration -I /home/opam/.opam/4.13/lib/easy-format -I /home/opam/.opam/4.13/lib/eqaf -I /home/opam/.opam/4.13/lib/eqaf/bigstring -I /home/opam/.opam/4.13/lib/eqaf/cstruct -I /home/opam/.opam/4.13/lib/fmt -I /home/opam/.opam/4.13/lib/fpath -I /home/opam/.opam/4.13/lib/gitlab -I /home/opam/.opam/4.13/lib/gitlab-unix -I /home/opam/.opam/4.13/lib/inotify -I /home/opam/.opam/4.13/lib/inotify/lwt -I /home/opam/.opam/4.13/lib/ipaddr -I /home/opam/.opam/4.13/lib/ipaddr-sexp -I /home/opam/.opam/4.13/lib/ipaddr/unix -I /home/opam/.opam/4.13/lib/irmin-watcher -I /home/opam/.opam/4.13/lib/ke -I /home/opam/.opam/4.13/lib/logs -I /home/opam/.opam/4.13/lib/lwt -I /home/opam/.opam/4.13/lib/lwt-dllist -I /home/opam/.opam/4.13/lib/lwt/unix -I /home/opam/.opam/4.13/lib/macaddr -I /home/opam/.opam/4.13/lib/magic-mime -I /home/opam/.opam/4.13/lib/mirage-crypto -I /home/opam/.opam/4.13/lib/mirage-crypto-rng -I /home/opam/.opam/4.13/lib/mirage-crypto-rng/unix -I /home/opam/.opam/4.13/lib/multipart_form -I /home/opam/.opam/4.13/lib/multipart_form-lwt -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.13/lib/ocaml/compiler-libs -I /home/opam/.opam/4.13/lib/ocaml/threads -I /home/opam/.opam/4.13/lib/ocplib-endian -I /home/opam/.opam/4.13/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.13/lib/parsexp -I /home/opam/.opam/4.13/lib/pecu -I /home/opam/.opam/4.13/lib/ppx_derivers -I /home/opam/.opam/4.13/lib/ppx_deriving/api -I /home/opam/.opam/4.13/lib/ppx_deriving/runtime -I /home/opam/.opam/4.13/lib/ppx_deriving_yojson -I /home/opam/.opam/4.13/lib/ppx_deriving_yojson/runtime -I /home/opam/.opam/4.13/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.13/lib/ppxlib -I /home/opam/.opam/4.13/lib/ppxlib/ast -I /home/opam/.opam/4.13/lib/ppxlib/astlib -I /home/opam/.opam/4.13/lib/ppxlib/print_diff -I /home/opam/.opam/4.13/lib/ppxlib/stdppx -I /home/opam/.opam/4.13/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.13/lib/prettym -I /home/opam/.opam/4.13/lib/prometheus -I /home/opam/.opam/4.13/lib/prometheus-app -I /home/opam/.opam/4.13/lib/ptime -I /home/opam/.opam/4.13/lib/re -I /home/opam/.opam/4.13/lib/result -I /home/opam/.opam/4.13/lib/routes -I /home/opam/.opam/4.13/lib/rresult -I /home/opam/.opam/4.13/lib/seq -I /home/opam/.opam/4.13/lib/session -I /home/opam/.opam/4.13/lib/session-cohttp -I /home/opam/.opam/4.13/lib/session-cohttp-lwt -I /home/opam/.opam/4.13/lib/sexplib -I /home/opam/.opam/4.13/lib/sexplib0 -I /home/opam/.opam/4.13/lib/sqlite3 -I /home/opam/.opam/4.13/lib/stdlib-shims -I /home/opam/.opam/4.13/lib/stringext -I /home/opam/.opam/4.13/lib/tyxml -I /home/opam/.opam/4.13/lib/tyxml/functor -I /home/opam/.opam/4.13/lib/unstrctrd -I /home/opam/.opam/4.13/lib/unstrctrd/parser -I /home/opam/.opam/4.13/lib/uri -I /home/opam/.opam/4.13/lib/uri-sexp -I /home/opam/.opam/4.13/lib/uri/services -I /home/opam/.opam/4.13/lib/uutf -I /home/opam/.opam/4.13/lib/yojson -intf-suffix .ml -no-alias-deps -open Current_gitlab__ -o plugins/gitlab/.current_gitlab.objs/byte/current_gitlab__Api.cmo -c -impl plugins/gitlab/api.pp.ml)
# File "plugins/gitlab/api.ml", line 431, characters 23-27:
# 431 |           String.equal hash br.branch_full_commit.commit_id
#                              ^^^^
# Error: This expression has type string option
#        but an expression was expected of type String.t = string
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlopt.opt -w -40 -g -I plugins/gitlab/.current_gitlab.objs/byte -I plugins/gitlab/.current_gitlab.objs/native -I /home/opam/.opam/4.13/lib/ISO8601 -I /home/opam/.opam/4.13/lib/angstrom -I /home/opam/.opam/4.13/lib/ansi -I /home/opam/.opam/4.13/lib/asetmap -I /home/opam/.opam/4.13/lib/astring -I /home/opam/.opam/4.13/lib/atdgen -I /home/opam/.opam/4.13/lib/atdgen-runtime -I /home/opam/.opam/4.13/lib/base/caml -I /home/opam/.opam/4.13/lib/base64 -I /home/opam/.opam/4.13/lib/base64/rfc2045 -I /home/opam/.opam/4.13/lib/bigarray-overlap -I /home/opam/.opam/4.13/lib/bigstringaf -I /home/opam/.opam/4.13/lib/biniou -I /home/opam/.opam/4.13/lib/bos -I /home/opam/.opam/4.13/lib/bytes -I /home/opam/.opam/4.13/lib/camlp-streams -I /home/opam/.opam/4.13/lib/cmdliner -I /home/opam/.opam/4.13/lib/cohttp -I /home/opam/.opam/4.13/lib/cohttp-lwt -I /home/opam/.opam/4.13/lib/cohttp-lwt-unix -I /home/opam/.opam/4.13/lib/conduit -I /home/opam/.opam/4.13/lib/conduit-lwt -I /home/opam/.opam/4.13/lib/conduit-lwt-unix -I /home/opam/.opam/4.13/lib/cstruct -I /home/opam/.opam/4.13/lib/csv -I /home/opam/.opam/4.13/lib/current -I /home/opam/.opam/4.13/lib/current/cache -I /home/opam/.opam/4.13/lib/current/term -I /home/opam/.opam/4.13/lib/current_git -I /home/opam/.opam/4.13/lib/current_incr -I /home/opam/.opam/4.13/lib/current_web -I /home/opam/.opam/4.13/lib/domain-name -I /home/opam/.opam/4.13/lib/duration -I /home/opam/.opam/4.13/lib/easy-format -I /home/opam/.opam/4.13/lib/eqaf -I /home/opam/.opam/4.13/lib/eqaf/bigstring -I /home/opam/.opam/4.13/lib/eqaf/cstruct -I /home/opam/.opam/4.13/lib/fmt -I /home/opam/.opam/4.13/lib/fpath -I /home/opam/.opam/4.13/lib/gitlab -I /home/opam/.opam/4.13/lib/gitlab-unix -I /home/opam/.opam/4.13/lib/inotify -I /home/opam/.opam/4.13/lib/inotify/lwt -I /home/opam/.opam/4.13/lib/ipaddr -I /home/opam/.opam/4.13/lib/ipaddr-sexp -I /home/opam/.opam/4.13/lib/ipaddr/unix -I /home/opam/.opam/4.13/lib/irmin-watcher -I /home/opam/.opam/4.13/lib/ke -I /home/opam/.opam/4.13/lib/logs -I /home/opam/.opam/4.13/lib/lwt -I /home/opam/.opam/4.13/lib/lwt-dllist -I /home/opam/.opam/4.13/lib/lwt/unix -I /home/opam/.opam/4.13/lib/macaddr -I /home/opam/.opam/4.13/lib/magic-mime -I /home/opam/.opam/4.13/lib/mirage-crypto -I /home/opam/.opam/4.13/lib/mirage-crypto-rng -I /home/opam/.opam/4.13/lib/mirage-crypto-rng/unix -I /home/opam/.opam/4.13/lib/multipart_form -I /home/opam/.opam/4.13/lib/multipart_form-lwt -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.13/lib/ocaml/compiler-libs -I /home/opam/.opam/4.13/lib/ocaml/threads -I /home/opam/.opam/4.13/lib/ocplib-endian -I /home/opam/.opam/4.13/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.13/lib/parsexp -I /home/opam/.opam/4.13/lib/pecu -I /home/opam/.opam/4.13/lib/ppx_derivers -I /home/opam/.opam/4.13/lib/ppx_deriving/api -I /home/opam/.opam/4.13/lib/ppx_deriving/runtime -I /home/opam/.opam/4.13/lib/ppx_deriving_yojson -I /home/opam/.opam/4.13/lib/ppx_deriving_yojson/runtime -I /home/opam/.opam/4.13/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.13/lib/ppxlib -I /home/opam/.opam/4.13/lib/ppxlib/ast -I /home/opam/.opam/4.13/lib/ppxlib/astlib -I /home/opam/.opam/4.13/lib/ppxlib/print_diff -I /home/opam/.opam/4.13/lib/ppxlib/stdppx -I /home/opam/.opam/4.13/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.13/lib/prettym -I /home/opam/.opam/4.13/lib/prometheus -I /home/opam/.opam/4.13/lib/prometheus-app -I /home/opam/.opam/4.13/lib/ptime -I /home/opam/.opam/4.13/lib/re -I /home/opam/.opam/4.13/lib/result -I /home/opam/.opam/4.13/lib/routes -I /home/opam/.opam/4.13/lib/rresult -I /home/opam/.opam/4.13/lib/seq -I /home/opam/.opam/4.13/lib/session -I /home/opam/.opam/4.13/lib/session-cohttp -I /home/opam/.opam/4.13/lib/session-cohttp-lwt -I /home/opam/.opam/4.13/lib/sexplib -I /home/opam/.opam/4.13/lib/sexplib0 -I /home/opam/.opam/4.13/lib/sqlite3 -I /home/opam/.opam/4.13/lib/stdlib-shims -I /home/opam/.opam/4.13/lib/stringext -I /home/opam/.opam/4.13/lib/tyxml -I /home/opam/.opam/4.13/lib/tyxml/functor -I /home/opam/.opam/4.13/lib/unstrctrd -I /home/opam/.opam/4.13/lib/unstrctrd/parser -I /home/opam/.opam/4.13/lib/uri -I /home/opam/.opam/4.13/lib/uri-sexp -I /home/opam/.opam/4.13/lib/uri/services -I /home/opam/.opam/4.13/lib/uutf -I /home/opam/.opam/4.13/lib/yojson -intf-suffix .ml -no-alias-deps -open Current_gitlab__ -o plugins/gitlab/.current_gitlab.objs/native/current_gitlab__Api.cmx -c -impl plugins/gitlab/api.pp.ml)
# File "plugins/gitlab/api.ml", line 431, characters 23-27:
# 431 |           String.equal hash br.branch_full_commit.commit_id
#                              ^^^^
# Error: This expression has type string option
#        but an expression was expected of type String.t = string
```

benmandrew commented:
The issue was related to an out-of-date vendored submodule. I've removed it and corrected the code upstream.

I'll close this PR to skip the release of 0.6.5 and resubmit using the patch as 0.6.6 #25181.
